src/angular/hq/src/app/clients/client-details/client-details-summary/client-details-summary.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router, RouterLink, RouterLinkActive, RouterModule, RouterOutlet } from '@angular/router';
+import { firstValueFrom, switchMap, of, Observable, catchError, map } from 'rxjs';
+import { HQService } from '../../../services/hq.service';
+import { GetClientRecordV1 } from '../../../models/clients/get-client-v1';
+import { APIError } from '../../../errors/apierror';
+
+@Component({
+  selector: 'hq-client-details-summary',
+  standalone: true,
+  imports: [],
+  templateUrl: './client-details-summary.component.html'
+})
+export class ClientDetailsSummaryComponent implements OnInit {
+    clientId?: string;
+    client?: GetClientRecordV1;
+    apiErrors: string[] = [];
+
+    async ngOnInit(): Promise<void> {
+      this.route.paramMap.pipe(
+        switchMap(params => {
+          this.clientId = params.get('clientId') || undefined;
+          return this.getClient().pipe(
+            catchError(error => {
+              if (error instanceof APIError) {
+                this.apiErrors = error.errors;
+              } else {
+                this.apiErrors = ['An unexpected error has occurred.'];
+              }
+              return of(null);
+            })
+          );
+        })
+      ).subscribe(client => {
+        this.client = client ?? undefined;
+      });
```

Comment:
Instead of subscribing and assigning `this.client`. Make an Observable<..> client$`  property and use an `| async` pipe in the template.